modules/core/src/test/kotlin/datamaintain/core/step/ExecutorTest.kt

```diff
@@ -189,4 +186,55 @@ internal class ExecutorTest {
                     }
         }
     }
+
+    @Test
+    fun `should execute and produce executed scripts with flags`() {
+        // Given
+        val context = Context(
+            DatamaintainConfig(Paths.get(""), Regex(""), driverConfig = FakeDriverConfig(), flags = listOf("FLAG1", "FLAG2", "FLAG3")),
+            dbDriver = dbDriverMock
+        )
+
+        val executor = Executor(context)
+
+        every { dbDriverMock.executeScript(any()) }.answers { Execution(OK) }
+        every { dbDriverMock.markAsExecuted(any()) }.answers { it.invocation.args.first() as ExecutedScript }
+
+        // When
+        val report = executor.execute(listOf(script1, script2, script3))
+
+        // Then
+        verify(exactly = 3) { dbDriverMock.executeScript(any()) }
+        verify(exactly = 3) { dbDriverMock.markAsExecuted(any()) }
```

Comment:
The point of this test is not to check that scripts were executed but to check if the correct flags were written during the execution. Thus, those two lines seem out of place and it's the same for the checks on execution statuses and even checks on names. What you want to ensure is that all scripts executed with such a configuration do have the given flags, not matter everything else.